### ISSUE TITLE
train ffhq

### ISSUE THREAD

**Issue by ljiqy** [omertov/encoder4editing#43]
Hi, thanks for the excellent work. I need to train my own face dataset, Can you provide scripts for training FFHQ(Some hyperparameters setting)?
In addition, my face image is 256x256, does it affect the model?

**Comment by omertov** [comment]
Hi @ljiqy!
To make sure I undestand correctly - you wish to train the e4e encoder based on your own dataset, with the pretrained FFHQ StyleGAN2?
If so, i suggest using the training script in the README: https://github.com/omertov/encoder4editing#training-the-e4e-encoder, 
with the following hyper parameters: `--max_steps 300000 --id_lambda 0.1`.
You should also specify `--stylegan_size 1024` (this is so the StyleGAN model will load properly), and change the `dataset_type` according to your dataset configuration (see https://github.com/omertov/encoder4editing#training-on-your-own-dataset).
It is fine that your images are 256x256, as we downscale the output 1024x1024 stylegan images into 256x256 resolution before calculating the loss.

In case you meant for training the StyleGAN itself, you should head over to <a href="https://github.com/rosinality/stylegan2-pytorch">rosinality's stylegan2-pytorch repo</a> and train the StyleGAN there before importing the trained checkpoint to be used in this repo.

Hope it helps!
Omer

**Comment by ljiqy** [comment]
Thank you for your reply quickly!
I have finish the stylegan2 training according rosinality's stylegan2-pytorch, now I use this script to train e4e:
python scripts/train.py \
--dataset_type my_ffhqencode \
--exp_dir ./my_model/mye4e \
--start_from_latent_avg \
--use_w_pool \
--w_discriminator_lambda 0.1 \
--progressive_start 20000 \
--id_lambda 0.5 \
--val_interval 10000 \
--max_steps 250000 \
--stylegan_size 1024 \
--stylegan_weights ./pretrained_models/my-trained-stylegan2-config-f.pt \
--workers 8 \
--batch_size 8 \
--test_batch_size 4 \
--test_workers 4

after using this script training, it doesn't work very well, If I understand you correctly, I need to change: --max_steps 300000 --id_lambda 0.1?

**Comment by omertov** [comment]
Yes, this is the configuration we ran for the FFHQ StyleGAN.
Can you elaborate on what's wrong with your results? Is it the reconstruction quality or the editing?

**Comment by ljiqy** [comment]
the reconstruction quality problem，I use twenty thousand images to train, my result(above is the original image, below is the reconstruction):
![7_ori](https://user-images.githubusercontent.com/37796840/126420400-9d532551-738a-4853-bd0f-767533e7d514.jpg)
![00004](https://user-images.githubusercontent.com/37796840/126420432-bf553c62-66e5-4a1c-9574-fa5ef5c96653.jpg)

**Comment by omertov** [comment]
Hi @ljiqy,
Our encoder is known for favoring editing properties by giving up on slight reconstruction (distortion) quality, so perhaps this is where the reconstruction degradation comes from.

Other possibilities for such degradation might come from the GAN itself - perhaps the "well-behaved" regions in the GAN's latent space are not expressive enough for generating the source image, so it opts to the given result as the closest image (I wonder what happens when you generate a random image and then encode it back into the latent space).
Another way of testing it is by training a regular pSp network (removing the `--w_discriminator_lambda --progressive_start` keys and adding `--encoder_type GradualStyleEncoder`) to see if the resulting image is better.
Also it is possible that more training steps will help.

Best,
Omer
